spec/unit/pushprocessor.spec.ts

```diff
@@ -188,6 +198,108 @@ describe("NotificationService", function () {
     });
 
     it("should add default rules in the correct order", () => {
+        matrixClient.pushRules = PushProcessor.rewriteDefaultRules({
```

Comment:
I'm really confused about what's going on here. Why are we overriding the push rules that we've just set up in the `beforeEach`? Doesn't calling `rewriteDefaultRules` twice invalidate the rest of the test? What is special about this set of push rules compared to what we'd have normally (write a comment)?